cmd/template/cluster/provider/aws.go

```diff
@@ -2,19 +2,139 @@ package provider
 
 import (
 	"io"
+	"os"
 	"text/template"
 
 	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
 	"github.com/giantswarm/apiextensions/v3/pkg/apis/infrastructure/v1alpha2"
 	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	capav1alpha3 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
 	"sigs.k8s.io/yaml"
 
 	"github.com/giantswarm/kubectl-gs/internal/key"
+	"github.com/giantswarm/kubectl-gs/internal/label"
 )
 
 func WriteAWSTemplate(out io.Writer, config ClusterCRsConfig) error {
 	var err error
 
+	if key.IsCAPAVersion(config.ReleaseVersion) {
+		err = WriteCAPATemplate(out, config)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	} else {
+		err = WriteGSAWSTemplate(out, config)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func WriteCAPATemplate(out io.Writer, config ClusterCRsConfig) error {
+	var err error
+
+	if config.ExternalSNAT {
+		return microerror.Maskf(invalidFlagError, "--external-snat setting is not available for release %v", config.ReleaseVersion)
+	}
+	if config.PodsCIDR != "" {
+		return microerror.Maskf(invalidFlagError, "--pods-cidr setting is not available for release %v", config.ReleaseVersion)
+	}
+
+	clusterTemplate, err := getCAPAClusterTemplate(config)
+	if err != nil {
+		return err
+	}
+
+	data := struct {
+		AWSClusterCR             string
+		AWSMachineTemplateCR     string
+		ClusterCR                string
+		KubeadmControlPlaneCR    string
+		AWSClusterRoleIdentityCR string
+	}{}
+
+	crLabels := map[string]string{
+		label.ReleaseVersion:            config.ReleaseVersion,
+		label.Cluster:                   config.ClusterID,
+		"cluster.x-k8s.io":              config.ClusterID,
```

Comment:
On the Azure implementation we use `cluster.x-k8s.io/cluster-name`.
https://github.com/giantswarm/kubectl-gs/blob/4027733d3b3954de75f8535e19b3bb32c11f810e/cmd/template/cluster/provider/azure.go#L81

Is the different label used on purpose?

Reply:
I'll check but I think this is just a typo. good catch.